app/actions/local/draft.ts

```diff
@@ -197,3 +212,99 @@ export async function updateDraftPriority(serverUrl: string, channelId: string,
         return {error};
     }
 }
+
+export async function updateDraftMarkdownImageMetadata({
+    serverUrl,
+    channelId,
+    rootId,
+    imageMetadata,
+    prepareRecordsOnly = false,
+}: {
+    serverUrl: string;
+    channelId: string;
+    rootId: string;
+    imageMetadata: Dictionary<PostImage | undefined>;
+    prepareRecordsOnly?: boolean;
+}) {
+    try {
+        const {database, operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const draft = await getDraft(database, channelId, rootId);
+        if (draft) {
+            draft.prepareUpdate((d) => {
+                d.metadata = {
+                    ...d.metadata,
+                    images: imageMetadata,
+                };
+                d.updateAt = Date.now();
+            });
+            if (!prepareRecordsOnly) {
+                await operator.batchRecords([draft], 'updateDraftImageMetadata');
+            }
+        }
+        return {draft};
+    } catch (error) {
+        logError('Failed updateDraftImages', error);
+        return {error};
+    }
+}
+
+async function getImageMetadata(url: string) {
+    let height = 0;
+    let width = 0;
+    let format;
+    try {
+        await new Promise((resolve, reject) => {
+            Image.getSize(
+                url,
+                (imageWidth, imageHeight) => {
+                    width = imageWidth;
+                    height = imageHeight;
+                    resolve(null);
+                },
+                (error) => {
+                    logError('Failed to get image size', error);
+                    reject(error);
+                },
+            );
+        });
+    } catch (error) {
+        width = 0;
+        height = 0;
```

Comment:
Do we need this?

The width and height are set to 0 already.  What about the error?  Do we need try catch here?

Reply:
We don't need try catch. Removed it.